application/src/objects/obstacles/abstract-dynamic-obstacle.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+var Phaser = require('phaser');
+
+var AbstractDynamicObstacle = function(state, x, y, key, collisionGroup)
+{
+    Phaser.Sprite.apply(this, [state.game, x, y, key]);
+
+    this.createPhysicsBody(state);
+
+    this.body.setCollisionGroup(collisionGroup);
+    this.body.mass = this.getAttributes().mass;
+    this.body.angularDamping = this.getAttributes().angularDamping;
+};
+
+AbstractDynamicObstacle.prototype = Object.create(Phaser.Sprite.prototype);
+
+AbstractDynamicObstacle.prototype.getSpritePath = function()
+{
+    throw new Error('Attempted to load assets on abstract class');
+};
+
+AbstractDynamicObstacle.prototype.getAttributes = function()
```

Comment:
I solved this problem differently (see 759927d).  I think I might prefer your function approach to my variables, but how do you feel about renaming it to getConstants? Seems more descriptive than "attributes". Also, how do you feel about using CONSTANT_CASE. It might not be very javascripty, but it makes it clear that it's a value that isn't meant to be changed ever.